src/runtime/stores/locale.ts

```diff
@@ -77,7 +77,8 @@ $locale.set = (newLocale: string): void | Promise<void> => {
 };
 
 // istanbul ignore next
-$locale.update = (fn: (locale: string) => void | Promise<void>) =>
-  localeSet(fn(current));
+$locale.update = (
+  fn: (value: string | null | undefined) => string | null | undefined,
```

Comment:
`fn` should return void

Reply:
2.
What do you think as the usage of `.update`?

Type of `import('svelte/store').Writable<T>['update']` is `(updater: Updater<T>) => void`.
`Updater<T>` is `(v: T) => v`, it is representing automorphism. For example, we can use `.update` like following.

```ts
locale.update((old) => {
  // switch 'en' and 'ja'
  return old === 'en' ? 'ja' : 'en';
})
```

And I've realized that the reassignment in this file did not make any sence.

We can also see it is strange becase it uses `void` as value. `undefined` and `void` is identical, but `void` is type used for unsable type.